servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CelAccessChecker.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server.authz;
+
+import com.google.common.collect.ImmutableMap;
+import java.security.AccessControlException;
+import java.util.Map;
+import java.util.function.Supplier;
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import org.projectnessie.cel.tools.ScriptException;
+import org.projectnessie.model.ContentsKey;
+import org.projectnessie.server.config.QuarkusNessieAuthorizationConfig;
+import org.projectnessie.services.authz.AccessChecker;
+import org.projectnessie.services.authz.AccessContext;
+import org.projectnessie.versioned.NamedRef;
+
+/**
+ * A reference implementation of the {@link AccessChecker} that performs access checks using CEL
+ * expressions.
+ */
+@ApplicationScoped
+public class CelAccessChecker implements AccessChecker {
+  private final QuarkusNessieAuthorizationConfig config;
+  private final CompiledAuthorizationRules compiledRules;
+
+  public enum AuthorizationRuleType {
+    VIEW_REFERENCE,
+    CREATE_REFERENCE,
+    DELETE_REFERENCE,
+    LIST_COMMIT_LOG,
+    READ_ENTRIES,
+    ASSIGN_REFERENCE_TO_HASH,
+    COMMIT_CHANGE_AGAINST_REFERENCE,
```

Comment:
Is  the rule for COMMIT_CHANGE_AGAINST_REFERENCE the one that covers the check for a user allowed to do the "merge" operation ? i.e commit changes from one branch to another ?

Reply:
`COMMIT_CHANGE_AGAINST_REFERENCE` covers merge / transplant / commits